internal/platform/select_darwin.go

```diff
@@ -0,0 +1,56 @@
+//go:build darwin || unix
+
+package platform
+
+import (
+	"syscall"
+	"time"
+)
+
+func SelectTimeout(timeout time.Duration) error {
+	return Select(0, nil, nil, nil, timeout)
+}
+
+func SelectStdin(timeout time.Duration) (bool, error) {
+	fdSet := &FDSet{}
+	fdSet.Set(0)
+	err := Select(1, fdSet, nil, nil, timeout)
+	if err != nil {
+		return false, err
+	}
+	return fdSet.IsSet(0), nil
+}
+
+func Select(n int, r, w, e *FDSet, timeout time.Duration) error {
+	t := syscall.NsecToTimeval(timeout.Nanoseconds())
+	return syscall.Select(n, (*syscall.FdSet)(r), (*syscall.FdSet)(w), (*syscall.FdSet)(e), &t)
+}
+
+// Code lifted from https://github.com/creack/goselect (MIT licensed)
+// Ports the macros FD_SET, FD_CLEAR, etc. to methods
+
+// FDSet wraps syscall.FdSet with convenience methods
+type FDSet syscall.FdSet
```

Comment:
sorry I forgot to add the comment of an alternate suggestion.

It seems to me that the platform-specific stuff we are ending up in copy/paste between impls are around two things: the type syscall.FdSet and also the mask value of NFDBITS.

NFDBITS isn't something library specific to goselect, it was likely copied/derived from x/sys and in any case, let's model x/sys not other codebases e.g. https://github.com/golang/sys/blob/master/unix/fdset.go

one trick you can see they did nicely is to define a separate file for fdset (and fdset_test). This allows you to centralize the copy/paste stuff. you can also put a dummy value in for _windows (as windows isn't supported, but this lets you compile).